src/components/datepicker/datePicker.js

```diff
@@ -365,55 +535,108 @@
    * The 'required' flag is handled automatically by ngModel.
    *
    * @param {Date=} opt_date Date to check. If not given, defaults to the datepicker's model value.
+   * @param {Date=} opt_begin_date Date to check. If not given, defaults to the datepicker's begin date value.
+   * @param {Date=} opt_end_date Date to check. If not given, defaults to the datepicker's end date value.
    */
-  DatePickerCtrl.prototype.updateErrorState = function(opt_date) {
-    var date = opt_date || this.date;
+  DatePickerCtrl.prototype.updateErrorState = function(opt_date, opt_begin_date, opt_end_date) {
+    if (!this.isDateRangePicker) {
+      var date = opt_date || this.date;
 
-    // Clear any existing errors to get rid of anything that's no longer relevant.
-    this.clearErrorState();
+      // Clear any existing errors to get rid of anything that's no longer relevant.
+      this.clearErrorState();
 
-    if (this.dateUtil.isValidDate(date)) {
-      // Force all dates to midnight in order to ignore the time portion.
-      date = this.dateUtil.createDateAtMidnight(date);
+      if (this.dateUtil.isValidDate(date)) {
```

Comment:
There is a lot of logic duplicated in this method. It can be cleaned up a lot by taking the function as-is and simply grabbing the min date/max date from the appropriate field.